api/overdrive.py

```diff
@@ -528,19 +570,28 @@ def get_fulfillment_link(self, patron, pin, overdrive_id, format_type):
 
         if format_type and not download_link:
             download_link = self.get_download_link(
-                loan, format_type, self.DEFAULT_ERROR_URL)
+                loan, format_type, self.DEFAULT_ERROR_URL
+            )
             if not download_link:
                 raise CannotFulfill(
                     "No download link for %s, format %s" % (
                         overdrive_id, format_type))
 
         if download_link:
+            if format_type in self.MANIFEST_INTERNAL_FORMATS:
+                # The client must authenticate using its own
+                # credentials to fulfill this URL; we can't do it.
+                scope_string = self.scope_string(patron.library)
+                return OverdriveManifestFulfillmentInfo(
+                    self.collection, download_link,
+                    overdrive_id, scope_string
+                )
```

Comment:
Here's where we create an `OverdriveManifestFulfillmentInfo`, short-circuiting the rest of the fulfillment process.